### PR TITLE
Decouple dataset from statistics

### PR DESCRIPTION
**Pull request by mschwoer**
This is a huge one, but could not be separated in a sensible way.. 
check the individual commits for the line of reasoning (the bug in fc calculation in [de72c6e](https://github.com/MannLabs/alphapeptstats/pull/329/commits/de72c6e19ce4a459c2581a2ecd57a912e5b3854d) is removed again later

I think now I see where this could be going: the `DataSet` class will contain the data (`mat`, `metadata`) and beside that be just a convenient interface for the frontend. It will offer all operations that can be done on the data (statistics, plots, preprocessing), but delegate the logic to dedicated classes. Two things are missing still:
- decoupling from `Plots`
- moving the logic in `init` (i.e. the actual data loading) to another class (similar to `Statistics`, ..)

### CODE REVIEW

File: alphastats/DataSet_Statistics.py [hunk 1]
```diff
@@ -194,36 +130,36 @@ def ancova(
...
-            covariates=covariates,
-            n_permutations=n_permutations,
-            fdr=fdr,
-            s0=s0,
-            subset=subset,
-            plot=True,
-        ).calculate()
-        return res, plot_list
+    # @ignore_warning(RuntimeWarning)
+    # def multicova_analysis(  # TODO never used outside of tests .. how does this relate to multicova.py?
```
Comment (JuliaS92) [inline alphastats/DataSet_Statistics.py:134]: In MultiCovaAnalysis multicova.py is imported from the calculate function.
```
    def calculate(self):
        from alphastats.multicova import multicova
```

This is defenitely a feature we want in the tool, so it should remain active but we need a ticket for putting it in the interface.

File: alphastats/gui/utils/analysis_helper.py [hunk 2]
```diff
@@ -386,7 +386,3 @@ def load_options():
     st.session_state[StateKeys.STATISTIC_OPTIONS] = statistic_options(st.session_state)
     # TODO: Check if this should be reintroduced or removed
     # st.session_state["interpretation_options"] = interpretation_options
-
-
-def gui_multicova_analysis():
```
Comment (JuliaS92) [inline alphastats/gui/utils/analysis_helper.py:391]: Add big todo here instead of removing it.

File: alphastats/plots/VolcanoPlot.py [hunk 3]
```diff
@@ -152,7 +175,7 @@ def _sam_calculate_fdr_line(self):
         )
 
     @lru_cache(maxsize=20)
-    def _sam(self):
+    def _sam(self):  # TODO duplicated?
```
Comment (JuliaS92) [inline alphastats/plots/VolcanoPlot.py:178]: Only the last two lines differ from the function in DifferentialExpressionAnalysis

**Review by JuliaS92 — COMMENTED** [review]
LGTM

**Review by boopthesnoot — COMMENTED** [review]
LGTM

**Pull request merged**